c/tree.c

```diff
@@ -1145,8 +1153,15 @@ LOCAL void encode_node(MMDBW_tree_s *tree, MMDBW_node_s *node,
 
     uint32_t left =
         htonl(record_value_as_number(tree, &(node->left_record), args));
+    if (0 == left) {
```

Comment:
It's definitely possible for a node to refer to a lower numbered node because of the way the IP6 -> 4 aliasing works.

The check for 0 is better than nothing, but we could implement a full on check for cycles if we wanted to.